testsuite/integration/microprofile/src/test/java/org/wildfly/test/integration/microprofile/health/MicroProfileHealthDisabledDefaultProceduresTestBase.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright The WildFly Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.wildfly.test.integration.microprofile.health;
+
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.OperateOnDeployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.as.arquillian.api.ContainerResource;
+import org.jboss.as.arquillian.container.ManagementClient;
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.asset.StringAsset;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.io.IOException;
+import java.util.List;
+
+@RunWith(Arquillian.class)
+@RunAsClient
+public abstract class MicroProfileHealthDisabledDefaultProceduresTestBase<CHECK_TYPE extends Object> {
+
+    protected abstract List<CHECK_TYPE> retrieveChecks(ManagementClient managementClient, final String checkType) throws IOException;
+
+    @Deployment(name = "MicroProfileHealthDisabledDefaultProceduresTestBase")
+    public static Archive<?> deployDisabledDefaultProcedures() {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, "DisabledProcedures.war")
+                .addClass(MyReadyProbe.class)
+                .addAsManifestResource(new StringAsset("mp.health.disable-default-procedures=true"),
+                    "microprofile-config.properties")
+                .addAsWebInfResource(new StringAsset("<beans bean-discovery-mode=\"all\"></beans>"), "beans.xml");
+        return war;
+    }
+
+    @ContainerResource
+    ManagementClient managementClient;
+
+    @Test
+    @OperateOnDeployment("MicroProfileHealthDisabledDefaultProceduresTestBase")
+    public void testDisabledDefaultProcedures() throws IOException {
+        // we expect one
+        final List<CHECK_TYPE> readinessChecks = retrieveChecks(managementClient, "ready");
+        // we expect none
+        final List<CHECK_TYPE> livenessChecks = retrieveChecks(managementClient, "live");
+        // we expect none
+        final List<CHECK_TYPE> startedChecks = retrieveChecks(managementClient, "started");
+
+        // let's just get the total checks count
+        Assert.assertEquals(1, readinessChecks.size() + livenessChecks.size(), startedChecks.size());
+        // then we expect the only one we find, implemented by the application level Health Check is returning "UP"
+
+        testReadinessChecks(readinessChecks);
+    }
+
+    protected void testReadinessChecks(final List<CHECK_TYPE> readinessChecks) throws IOException {
```

Comment:
and you don't need this as it doesn't test anything that you already didn't test

Reply:
Ok. Removed this implementation by making the method abstract.